clients/src/main/java/org/apache/kafka/common/protocol/types/CompactArrayOf.java

```diff
@@ -119,7 +119,7 @@ public Object[] validate(Object item) {
                 type.validate(obj);
             return array;
         } catch (ClassCastException e) {
-            throw new SchemaException("Not an Object[].");
+            throw new SchemaException("Not an Object[].", e);
```

Comment:
I think it would be more helpful to include the class of `item` than chaining the exception here.

Reply:
I have fixed it as per your suggestion.